peer/doc.go

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2013-2015 The btcsuite developers
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+// Package peer provides a common base for creating and managing bitcoin network
+// peers for fully validating, Simplified Payment Verification (SPV) nodes,
+// proxies etc. It handles all the common message handlers like initial message
+// version negotiation and handling and responding to pings. For other messages,
+// it provides a mechanism to register and manage custom handlers which will be
+// invoked on receiving the given message.
+
+// Creating a peer requires a method to retrieve the newest block sha, a nonce,
+// a channel to signal peer completion, an address manager and customizable
+// config. Once initialized, a peer can be started using the Start method.
+
```

Comment:
There can't be a space here or the documentation will not show up.  I'd suggest using block comments as well for consistency.

```Go
/*
Package peer ...

Creating a peer ....
*/
package peer
```